pkg/master/master_test.go

```diff
@@ -363,6 +377,108 @@ func TestAPIVersionOfDiscoveryEndpoints(t *testing.T) {
 
 }
 
+func TestStorageVersionHashExist(t *testing.T) {
+	defer utilfeaturetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.StorageVersionHash, true)()
+	exceptions := sets.NewString(
+		"bindings",
+		"componentstatuses",
+		"tokenreviews",
+		"localsubjectaccessreviews",
+		"selfsubjectaccessreviews",
+		"selfsubjectrulesreviews",
+		"subjectaccessreviews",
+		"extensions/v1beta1/replicationcontrollers",
+	)
+	master, etcdserver, _, _ := newMaster(t)
+	defer etcdserver.Terminate(t)
+
+	server := httptest.NewServer(master.GenericAPIServer.Handler.GoRestfulContainer.ServeMux)
+
+	c := &restclient.Config{
+		Host:          server.URL,
+		APIPath:       "/api",
+		ContentConfig: restclient.ContentConfig{NegotiatedSerializer: legacyscheme.Codecs},
+	}
+	discover := discovery.NewDiscoveryClientForConfigOrDie(c)
+	all, err := discover.ServerResources()
+	if err != nil {
+		t.Error(err)
+	}
+	for _, g := range all {
+		for _, r := range g.APIResources {
+			if strings.Contains(r.Name, "/") || exceptions.Has(r.Name) || exceptions.Has(g.GroupVersion+"/"+r.Name) {
+				if r.StorageVersionHash != "" {
+					t.Errorf("expect resource %q/%q to have empty storageVersionHash, got hash %q", g.GroupVersion, r.Name, r.StorageVersionHash)
+				}
+				continue
+			}
+			if r.StorageVersionHash == "" {
+				t.Errorf("expect all main resources to have storageVersionHash, but resource %q/%q has empty hash", g.GroupVersion, r.Name)
+			}
+		}
+	}
+}
+
+func TestStorageVersionHashEqualities(t *testing.T) {
+	defer utilfeaturetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.StorageVersionHash, true)()
+	master, etcdserver, _, assert := newMaster(t)
+	defer etcdserver.Terminate(t)
+
+	server := httptest.NewServer(master.GenericAPIServer.Handler.GoRestfulContainer.ServeMux)
+
+	// Test 1: extensions/v1beta1/replicasets and apps/v1/replicasets have
```

Comment:
This test seems valuable but not future proof-- maybe the prior test could be changed to make a map[hash][]GVK and then verify that every hash has just 1 entry except for these specific ones?

Reply:
It also seems important that these hashes don't change except when we change something about the storage; perhaps it would be good to check in a golden data file (somewhere owned by api reviewers) so that it's difficult to accidentally change this?

Reply:
I checked in the golden file at pkg/master/storageversionhashdata, and made api-reviewers as owners.